### ISSUE TITLE
bug: `am setup` writes Claude Code MCP config to `~/.claude/settings.json` (mcpServers) — a location Claude Code v2.x ignores → fresh `claude` instances silently get no Agent Mail tools

### ISSUE THREAD

**Issue by etafund** [Dicklesworthstone/mcp_agent_mail_rust#168]
## Summary

`am setup` configures Claude Code by writing the `mcp-agent-mail` server into **`~/.claude/settings.json`** (user) and **`<project>/.claude/settings.local.json`** (project-local) under an `mcpServers` key. **Current Claude Code (verified on v2.1.187) does not read MCP server definitions from `settings.json` / `settings.local.json` at all** — it only loads them from `~/.claude.json` (user + local scope) and project `.mcp.json`.

Net effect: after `am setup` (or after Claude Code auto-updates to a version that ignores `settings.json` `mcpServers`), **freshly launched `claude` instances silently have no Agent Mail tools** — `mcp__mcp-agent-mail__*` is simply absent. The MCP server itself is healthy and reachable; the client is just configured in a file Claude Code ignores. This is especially damaging for agent swarms, where every newly spawned `claude` pane is expected to have Agent Mail available the instant it boots.

This is a silent failure: `am setup` reports success, the JSON it writes is well-formed, and `am doctor` does not flag it — but `claude mcp list` shows **"No MCP servers configured"** and the tools never register.

## Root cause (exact location)

`crates/mcp-agent-mail-core/src/setup.rs`, `claude_actions()`:

```rust
// ~line 1004 — project-local
file_path: pdir.join(".claude").join("settings.local.json"),
content: ConfigContent::JsonMerge { servers_key: "mcpServers", server_name: "mcp-agent-mail", .. },

// ~line 1019 — user-level
file_path: home.join(".claude").join("settings.json"),
content: ConfigContent::JsonMerge { servers_key: "mcpServers", server_name: "mcp-agent-mail", .. },
```

Both write `mcpServers` into `settings.json` / `settings.local.json`. The detector side (`crates/mcp-agent-mail-core/src/mcp_config.rs:539-551`) also *reads back* from `~/.claude/settings.json`, `settings.local.json`, and `claude_desktop_config.json` — so `am setup` writes and `am`’s own detection both agree on a location that the **actual Claude Code runtime ignores**, which is why the misconfiguration is invisible to `am`.

Per Anthropic’s docs (https://code.claude.com/docs/en/mcp.md) and empirical testing, Claude Code reads MCP servers only from:
- `~/.claude.json` — top-level `mcpServers` (user scope) and `projects.<path>.mcpServers` (local scope)
- `<project>/.mcp.json` (project scope, checked into VCS)

`settings.json` is for permissions / hooks / env / UI — **not** MCP servers. (`claude_desktop_config.json` is the *Claude Desktop* app config, a different product, also not read by the Claude Code CLI.)

## Reproduction

```bash
# 1. Configure Claude Code via am setup (or hand-write the same shape am setup writes):
cat ~/.claude/settings.json   # contains: {"mcpServers": {"mcp-agent-mail": {"type":"http","url":"http://127.0.0.1:8765/mcp/","headers":{"Authorization":"Bearer ..."}}}}

# 2. Ask Claude Code what it sees:
claude mcp list
# -> "No MCP servers configured. Use `claude mcp add` to add a server."
claude mcp get mcp-agent-mail
# -> 'No MCP server named "mcp-agent-mail".'

# 3. Prove the tools never register in a fresh session:
claude -p "Call MCP tool mcp__mcp-agent-mail__health_check with {}. Print HEALTHLEVEL=<health_level>." --dangerously-skip-permissions
# -> "the MCP tool ... is not available in this session"

# Meanwhile the server is perfectly healthy and reachable:
curl -s -X POST -H 'Authorization: Bearer <token>' -H 'Content-Type: application/json' \
  -H 'Accept: application/json, text/event-stream' \
  http://127.0.0.1:8765/mcp/ \
  -d '{"jsonrpc":"2.0","id":1,"method":"tools/call","params":{"name":"health_check","arguments":{}}}'
# -> {"health_level":"green", ...}
```

## Fix (verified)

Re-registering the same server in the canonical Claude Code location makes it load immediately:

```bash
claude mcp add --transport http --scope user mcp-agent-mail \
  http://127.0.0.1:8765/mcp/ \
  --header "Authorization: Bearer <token>"

claude mcp list
# -> mcp-agent-mail: http://127.0.0.1:8765/mcp/ (HTTP) - ✔ Connected

claude -p "Call mcp__mcp-agent-mail__health_check with {}. Print HEALTHLEVEL=<health_level>." --dangerously-skip-permissions
# -> HEALTHLEVEL=green
```

`--scope user` persists to `~/.claude.json` top-level `mcpServers`, which auto-loads in both interactive and `-p` print mode across all working directories.

### Suggested change to `am setup` (`claude_actions`)

Target the locations Claude Code actually reads, instead of `settings.json`:

1. **User scope** → write/merge `mcpServers` into **`~/.claude.json`** (top-level `mcpServers`), not `~/.claude/settings.json`. Equivalent to `claude mcp add --transport http --scope user ...`.
2. **Project scope** → write/merge into **`<project>/.mcp.json`** (top-level `mcpServers`), not `<project>/.claude/settings.local.json`.
3. Keep writing **hooks** to `settings.json` (that part is correct — hooks *do* live in settings.json).
4. Update the read-back detector (`mcp_config.rs`) to inspect `~/.claude.json` (user + `projects.<path>`) and project `.mcp.json`, so `am setup`/`am doctor` can verify the config Claude Code will actually use. Optionally add a `doctor` check that flags a stray `mcpServers` block in `settings.json` (the now-ignored location) and offers to migrate it.

### Backward-compat note

This likely worked on older Claude Code builds (or was masked by a manual `--mcp-config`/`.mcp.json`), and broke silently when Claude Code auto-updated. A migration in `am setup`/`am doctor` (move `settings.json` `mcpServers` → `~/.claude.json` / `.mcp.json`, then drop the dead block) would self-heal existing installs.

## Impact

- Severity: **high for Claude Code users / swarms**. Every fresh `claude` instance launches with **zero** Agent Mail tools after a Claude Code update, despite a healthy server and a "successful" `am setup`. Agents cannot reserve files or send/receive mail until reconfigured, defeating the multi-agent coordination the project exists to provide.
- Silent: no error from `am setup`, `am doctor`, or Claude Code itself — only `claude mcp list` reveals it.
- Codex is unaffected (it correctly reads `~/.codex/config.toml [mcp_servers.*]`, and `codex mcp list` shows the server enabled + connected).

## Environment

- `am` 0.3.14 (built from source @ `main`)
- Claude Code 2.1.187
- Linux; server: `am serve-http --no-tui --host 127.0.0.1 --port 8765 --path /mcp/` under a systemd user unit, `health_level: green`.

**Comment by samxu01** [comment]
The "freshly spawned claude pane silently has no tools" problem is the kind of per-instance config fragility that goes away when identity and the shared channel live server-side instead of being wired into each pane on boot. We took that approach for agent-to-agent coordination — agents join a shared space rather than each booting with local config (Commonly, open-source — github.com/Team-Commonly/commonly). Curious how your swarm setup would map onto it.

**Comment by Dicklesworthstone** [comment]
Fixed on `main` in `87554e19`. `am setup` (`claude_actions`) now writes the Claude Code MCP server into `~/.claude.json` — top-level `mcpServers` (user scope) and `projects.<abs>.mcpServers` (local scope) via the new `ConfigContent::ClaudeLocalScopeMcp` / `merge_claude_local_scope_mcp` — which is where Claude Code v2.x actually reads MCP servers, instead of the ignored `~/.claude/settings.json` / `.claude/settings.local.json`. Hooks stay in `settings.json` (correct). The read-back detector (`mcp_config.rs`) now treats `~/.claude.json` + project `.mcp.json` as the primary Claude candidates (`preferred_config_path → ~/.claude.json`), keeping legacy `settings*.json` only for stale-config/migration detection — so `am setup`/`am doctor` verify the config Claude Code will actually use. As a bonus the token no longer lands in the project tree.
